src/main/java/de/retest/recheck/auth/RetestAuthentication.java

```diff
@@ -0,0 +1,231 @@
+package de.retest.recheck.auth;
+
+import static org.keycloak.adapters.rotation.AdapterTokenVerifier.verifyToken;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.UUID;
+
+import org.keycloak.OAuth2Constants;
+import org.keycloak.OAuthErrorException;
+import org.keycloak.adapters.KeycloakDeployment;
+import org.keycloak.adapters.KeycloakDeploymentBuilder;
+import org.keycloak.adapters.ServerRequest;
+import org.keycloak.adapters.ServerRequest.HttpFailure;
+import org.keycloak.adapters.rotation.AdapterTokenVerifier;
+import org.keycloak.common.VerificationException;
+import org.keycloak.common.util.KeycloakUriBuilder;
+import org.keycloak.representations.AccessToken;
+import org.keycloak.representations.AccessTokenResponse;
+
+import lombok.extern.slf4j.Slf4j;
+import okhttp3.HttpUrl;
+
+@Slf4j
+public class RetestAuthentication {
+	private static final String KEYCLOAK_JSON = "META-INF/keycloak.json";
+
+	private final KeycloakDeployment deployment;
+
+	private String refreshToken;
+	private AccessToken accessToken;
+	private String accessTokenString;
+
+	private static RetestAuthentication instance;
+
+	private RetestAuthentication() {
+		final InputStream config = Thread.currentThread().getContextClassLoader().getResourceAsStream( KEYCLOAK_JSON );
+		deployment = KeycloakDeploymentBuilder.build( config );
+	}
+
+	public static RetestAuthentication getInstance() {
+		if ( instance == null ) {
+			instance = new RetestAuthentication();
+		}
+		return instance;
+	}
+
+	public URI getAccountUrl() {
+		try {
+			return new URI( deployment.getAccountUrl() );
+		} catch ( final URISyntaxException e ) {
+			throw new IllegalStateException( "Error creating URI from " + deployment.getAccountUrl(), e );
+		}
+	}
+
+	public boolean isAuthenticated( final String offlineToken ) {
+		if ( offlineToken != null ) {
+			try {
+				final AccessTokenResponse response = ServerRequest.invokeRefresh( deployment, offlineToken );
+				accessToken = verifyToken( response.getToken(), deployment );
+				accessTokenString = response.getToken();
+				return true;
+			} catch ( IOException | HttpFailure e ) {
+				log.error( "Error verifying offline token", e );
+			} catch ( final VerificationException e ) {
+				log.info( "Token not recognized, please authenticate" );
```

Comment:
Level `INFO` and not passing `e` on purpose?

Reply:
Yes, it's not clearly an error if the token is no longer valid. I think the huge stack trace could be misleading.